app/lib/frontend/static_files.dart

```diff
@@ -96,4 +99,24 @@ class StaticUrls {
       'download': downloadIcon,
     };
   }
+
+  Map get assets {
+    return _assets ??= {
+      'script_dart_js': _getCacheableStaticUrl('/v2/js/script.dart.js'),
+    };
+  }
+}
+
+/// Returns the URL of a static resource
+String _getCacheableStaticUrl(String relativePath) {
+  if (!relativePath.startsWith('/')) {
+    relativePath = '/$relativePath';
+  }
+  final String requestPath = '${staticUrls.staticPath}$relativePath';
+  final file = staticsCache.getFile(requestPath);
+  if (file == null) {
+    throw new Exception('Static resource not found: $relativePath');
```

Comment:
I don't think we want to throw here, since this is called when initializing `_assets` lazily, by renderLayoutPage(), which (I'm guessing) would result in a 500 error message thrown back to users.

Maybe log an error instead, and fall back to `requestPath`? Then they should get a nice 404 when they follow the src href instead.

Reply:
This exception is thrown mostly because it can act as an early-warning in tests: if we reference something that does not exists anymore, we'll get notified by the failed test.

Reply:
True, we'll pick it up early if this happens.